### ISSUE TITLE
All mandatory form fields must vocalize "required" as soon as they gain focus

### ISSUE THREAD

**Issue by developpeurweb** [SAP/spartacus#13678]
**Describe the bug**
- Consistently all throughout, none of the mandatory form fields are currently being vocalized as "Required" by any screen reader software.

**Tell us the version of Spartacus**
- Library version: 4.0.0

**To Reproduce**

1. Open any Screen Reader (VO, NVDA, JAWS).
2. Go to the Sign-In page
3. Tab to the email address field.
4. Hear the absence of vocalization about this field being required.
5. Tab to the password field, same issue.

**Expected behavior**
- All mandatory form fields must vocalize being "Required" as soon as the user focuses on them.
- Use: `aria-required="true"`

**Non-thorough list of UIs where this is needed:**

- Login
- Registration
- Coupon field
- Payment details
- My Account > Personal details
- My Account > Password
- My Account > Email Address

**Screenshots**
**Example of potential solution**

![Screen Shot 2021-09-09 at 7 59 38 PM 2](https://user-images.githubusercontent.com/15113219/132782475-ea17f892-8509-4c32-87d3-78e2e64ec8e5.png)

**Comment by 0xfaustian** [comment]
Fields that already have a “required” attribute will vocalize, aria attribute was not added to these

For reference, it has been added to the following form fields

Login form

Registration form

Coupon Field in Cart

Payment Form

My account -> Update password

My account -> Update personal details

My account -> Update email

Quick order cart

Cs-agent login form

Saved cart form

Guest register form and  checkout

Budget form

Unit address Form (org)

User change password (org)

Forgot password  and Reset password

Address form

Product Reviews

**Comment by KateChuen** [comment]
Tested locally (installation script using develop branch). 

- [x] Login form
- [x] Registration form
- [x] Coupon Field in Cart
- [x] My account -> Update password
- [x] My account -> Update personal details
- [x] My account -> Update email
- [x] My account -> Address Book
- [x] Quick order cart
- [x] Cs-agent login form
- [x] Saved cart form
- [x] Guest register form and checkout
- [x] Budget form
- [x] Unit address Form (org)
- [x] User change password (org)
- [x] Forgot password and Reset password
- [x] Checkout - Address form (Shipping)
- [x] Checkout - Address form (Billing)
- [x] Checkout - Payment Form
- [x] Product Reviews

Please see comments below for some places where it's still missing:
-> For drop down menus, it's missing the "Required": 
- When focus is on Country drop down menu, in address form
- When focus is on the Payment type drop down menu, in payment form
-> For review stars, it's missing the "Required":
- When focus is on the review stars for Product reviews
-> budget form: 
- when focus is on "start date" and "end date"
- when focus is on "unit" drop down menu
-> address form in units Org
- when focus is on City field

**Comment by KateChuen** [comment]
Opened ticket for the remaining 2 issues that will be fixed later:
review stars: https://github.com/SAP/spartacus/issues/15039
start and end date: https://github.com/SAP/spartacus/issues/15040

**Comment by KateChuen** [comment]
Tested on develop branch of spartacus using installation script
More extensive testing (on other screenreaders and devices) will be done with RC.

- When focus is on Country drop down menu, in address form  ✅
![shippingAddReq](https://user-images.githubusercontent.com/45409515/154533618-f6313a4d-b1a8-4ebb-8c2f-0f85f00c2a49.png)


- When focus is on the Payment type drop down menu, in payment form ✅
![paymentTypeReq](https://user-images.githubusercontent.com/45409515/154533871-9f49fb99-8545-425e-a6c3-ed8c041f9431.png)


- For review stars, it's missing the "Required": When focus is on the review stars for Product reviews ❌ -> https://github.com/SAP/spartacus/issues/15039

- budget form: when focus is on "start date" and "end date" ❌ -> https://github.com/SAP/spartacus/issues/15040

- budget form: when focus is on "unit" drop down menu ✅ 
![unitReq](https://user-images.githubusercontent.com/45409515/154534443-4ef0d1c0-27c5-43e1-9d80-e8e7d8d5da7a.png)

- address form in units Org: when focus is on City field ✅ 
![cityUnitReq](https://user-images.githubusercontent.com/45409515/154534710-ceb3f2fb-1d44-4900-8862-038e582430e9.png)
